### DISCUSSION TITLE
Welcome to CS342 Building for Digital Health!

### DISCUSSION THREAD

**Discussion by PSchmiedmayer** [CS342/.github#3] [Announcements]
# Welcome to CS342 Building for Digital Health!

We look forward to exploring the possibilities of Digital Health applications, development for Apple platforms, and the tools we develop at the Byers Center for Biodesign as part of our digital health initiatives.

Feel free to check out the class schedule at cs342.stanford.edu to get a good idea of the topics covered and the guest lecturers we have invited.
The first few weeks of the course will focus on understanding your project and learning about mobile application development, challenges in digital health systems, and several software frameworks that will make it easier for you to develop the first iteration of the envisioned system.
As the technical classes will be hands-on, we ask you to follow a few setup steps and do some reading before the course, so you have a setup ready for the optimal course experience when we do in-class coding exercises and start with the project work.

## GitHub

We will use GitHub as a software development platform for our assessments, projects, and project management. GitHub Discussions will provide a place to get help and help each other.

1. Please read through the following page about GitHub and GitHub Discussions to learn about the possibilities that GitHub provides.
   - **[GitHub](https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account) - We recommend Part 1 to Part 3.**
   - We ask you to **set up a free GitHub Account** if you do not yet have one. We ask you to use a recognizable name and a recognizable profile picture so we can associate your account with you for your class project and when you ask a question in the GitHub Discussions forum. **Please share the GitHub username in the form your have received in the email linking to this announcement.** 
   - [GitHub Discussions](https://docs.github.com/en/discussions): **Please read the [Quickstart](https://docs.github.com/en/discussions/quickstart) and [Collaborating using discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions) sections**. Feel free to explore more on the page if you are interested.
2. If you have any problems during the setup or following the tutorials, we ask you to use GitHub Discussions to [start a discussion thread in the setup category](https://github.com/orgs/CS342/discussions/categories/setup). Please provide as much detail as possible using the form-based setup. This will enable everyone to learn from setup challenges and might already include a solution for a problem you are encountering.
3. Git is a distributed version control management system that GitHub is based on. Please follow the following tutorials as a first guide to understanding git and the GitHub flow.
   - **[Git basics and Hello World](https://docs.github.com/en/get-started/quickstart/hello-world)**
   - **[The GitHub Flow](https://docs.github.com/en/get-started/quickstart/github-flow)**

## MacBook Setup

You will need access to a MacBook to do mobile application development for the class to follow along with the coding exercises in class, participate in the project development, and work on the assessments. If you do not have any way to obtain access to a MacBook, we ask you to reach out to use to discuss how we could enable you getting a MacBook for the class.

Please ensure that your MacBook meets the following criteria and that you install or update the software on your Mac accordingly.

### MacOS Ventura - 13.1 Or Newer 

The **MacBook needs to run macOS Ventura 13.1 or newer**. Please update to the [latest operating system version following the Apple-provided instructions](https://support.apple.com/en-us/HT201541).

You can verify that you run the latest macOS version by clicking on the Apple Logo on the top left of your screen and selecting “About this Mac”. You can see the macOS version number in the specs list under your MacBook picture.

### Xcode - 14.2 Or Newer

Xcode is the integrated development environment (IDE) we will work with. **You need to have Xcode 14.2 or later installed**. [You can install Xcode using the Mac AppStore](https://apps.apple.com/us/app/xcode/id497799835).

Please open Xcode and follow the instructions to finish the installation.

You can verify that you run the latest version of Xcode and everything is installed if you can see the “Welcome to Xcode” screen when you open Xcode showing 14.2 or newer as the version number: 

## Readings and Your First iOS App

Applications for the Apple ecosystem are written in the [Swift programming language](https://developer.apple.com/swift/). The framework for developing the user interface for mobile applications in Swift is called [SwiftUI](https://developer.apple.com/xcode/swiftui/). We will have a lecture about more advanced concepts of Swift and an overview of SwiftUI as part of lecture 1B.
Please go through the following readings and tutorials before the class to learn about the core language constructs and get a first understanding of SwiftUI to get an optimal experience in the course.

### The Swift Language Guide

The Swift Language Guide provides a comprehensive overview of the Swift programming language. **Please read and explore the ["A Swift Tour"](https://docs.swift.org/swift-book/GuidedTour/GuidedTour.html) in detail** to overview to get a good understanding of the core concepts of the programming language.

The [Swift Language Guide](https://docs.swift.org/swift-book/LanguageGuide/TheBasics.html) itself provides an in-depth introduction to Swift. We recommend using the language guide as a continuous reference in the course. While we do not require you to read through the Language Guide now, we require it as a reading for the first Assessment. You can already get a head-start if you have time before class.

We can also highly recommend [swiftbysundell.com](https://swiftbysundell.com) and [hackingwithswift.com](https://www.hackingwithswift.com) as great resources to follow up on questions and learn more about some concepts for the class. The [100 days of Swift](https://www.hackingwithswift.com/100) and [100 days of SwiftUI](https://www.hackingwithswift.com/100/swiftui) are amazing resources to learn more about Swift and SwiftUI during the course. The [Swift Forums](https://forums.swift.org). We can especially recommend the [Using Swift](https://forums.swift.org/c/swift-users/15) part of the forums as a great forum to ask Swift-related questions. The forums are also a wonderful place to ask questions about getting started with Swift, connect with the developers of the Swift programming language, and embrace the Swift community.

### Create your first Xcode Project

**Please follow the [Creating and Combining Views](https://developer.apple.com/tutorials/swiftui/creating-and-combining-views) tutorial** to test out your Xcode installation and try out the iOS Simulator so you can ensure that your setup is working as expected for the class. We can recommend taking a look at [Editing source files in Xcode](https://developer.apple.com/documentation/xcode/creating-organizing-and-editing-source-files) to learn more about the source editor in Xcode.

The first project you created above uses Swift and SwiftUI. We will build on top of that knowledge to teach you about the amazing opportunities of SwiftUI in the first week of class.

## Summary

You made it! We look forward to seeing you in person on our first day of class on January 10, 2023! If you run into any problems or have any challenges during the setup, please use the GitHub Discussions “Setup” section to ask your questions, look at existing questions that might cover the same problem, or help each other if you are already an experienced Mac user, Xcode user, or Swift expert!

**Comment by vishnuravi** [comment]
Welcome everyone!

**Comment by alimottaghi** [comment]
Excellent! Looking forward to starting strong! 💪🏻🔝

**Comment by sherryycxie** [comment]
Excited for our first class on Tuesday!
